pkg/cloudprovider/providers/aws/aws_loadbalancer.go

```diff
@@ -157,10 +157,10 @@ func (c *Cloud) ensureLoadBalancer(namespacedName types.NamespacedName, loadBala
 
 				found := -1
 				for i, expected := range listeners {
-					if orEmpty(actual.Protocol) != orEmpty(expected.Protocol) {
+					if elbProtocolsAreEqual(actual.Protocol, expected.Protocol) {
```

Comment:
This seems to invert the condition in addition to making it case-insensitive. Was that intentional?